pkg/provisioner/filestorage/filestorage.go

```diff
@@ -0,0 +1,218 @@
+		},
+	})
+	if err != nil {
+		glog.Errorf("Failed to create a file system storage:%#v, %s", options, err)
+		return nil, err
+	}
+
+	mntTargetResp := filestorage.MountTarget{}
+	if options.Parameters[mntTargetID] == "" {
+		// Check if there there already is a mount target in the existing compartment
+		glog.Infof("Looking up existing mount targets")
+		responseListMnt, err := fileStorageClient.ListMountTargets(ctx, filestorage.ListMountTargetsRequest{
+			AvailabilityDomain: availabilityDomain.Name,
+			CompartmentId:      common.String(filesystem.client.CompartmentOCID()),
+		})
+		if err != nil {
+			glog.Errorf("Failed to list mount targets:%#v, %s", options, err)
+			return nil, err
+		}
+		if len(responseListMnt.Items) != 0 {
+			glog.Infof("Found mount targets to use")
+			rand.Seed(time.Now().Unix())
+			mntTargetSummary := responseListMnt.Items[rand.Int()%len(responseListMnt.Items)]
+			mntTargetResp = *getMountTargetFromID(ctx, *mntTargetSummary.Id, fileStorageClient)
+		} else {
+			// Mount target not created, create a new one
+			responseMnt, err := fileStorageClient.CreateMountTarget(ctx, filestorage.CreateMountTargetRequest{
+				CreateMountTargetDetails: filestorage.CreateMountTargetDetails{
+					AvailabilityDomain: availabilityDomain.Name,
+					SubnetId:           common.String(options.Parameters[subnetID]),
+					CompartmentId:      common.String(filesystem.client.CompartmentOCID()),
+					DisplayName:        common.String(fmt.Sprintf("%s%s", os.Getenv(volumePrefixEnvVarName), "mnt")),
+				},
+			})
+			if err != nil {
+				glog.Errorf("Failed to create a mount target:%#v, %s", options, err)
+				return nil, err
+			}
+			mntTargetResp = responseMnt.MountTarget
+		}
+	} else {
+		// Mount target already specified in the configuration file, find it in the list of mount targets
+		mntTargetResp = *getMountTargetFromID(ctx, options.Parameters[mntTargetID], fileStorageClient)
+	}
+
+	glog.Infof("Creating export set")
+	createExportResponse, err := fileStorageClient.CreateExport(ctx, filestorage.CreateExportRequest{
+		CreateExportDetails: filestorage.CreateExportDetails{
+			ExportSetId:  mntTargetResp.ExportSetId,
+			FileSystemId: response.FileSystem.Id,
+			Path:         common.String("/" + *response.FileSystem.Id),
+		},
+	})
+
+	if err != nil {
+		glog.Errorf("Failed to create export:%s", err)
+		return nil, err
+	}
+	serverIP := ""
+	if len(mntTargetResp.PrivateIpIds) != 0 {
+		privateIPID := mntTargetResp.PrivateIpIds[rand.Int()%len(mntTargetResp.PrivateIpIds)]
+		virtualNetworkClient := filesystem.client.VirtualNetwork()
+		getPrivateIPResponse, err := virtualNetworkClient.GetPrivateIp(ctx, core.GetPrivateIpRequest{
+			PrivateIpId: common.String(privateIPID),
+		})
+		if err != nil {
+			glog.Errorf("Failed to retrieve IP address for mount target:%s", err)
+			return nil, err
+		}
+		serverIP = *getPrivateIPResponse.PrivateIp.IpAddress
+	} else {
```

Comment:
In Go the best practice is to handle the error case (`len(mntTargetResp.PrivateIpIds) == 0`) first and return to minimise indentation.

Reply:
done